torch/testing/_creation.py

```diff
@@ -0,0 +1,159 @@
+"""
+This package contains tensor creation utilities.
+"""
+
+import torch
+from typing import Optional, Tuple, Union
+import math
+
+__all__ = [
+    "make_tensor"
+]
+
+def make_tensor(
+    size: Tuple[int, ...],
+    *,
+    device: Union[str, torch.device] = "cpu",
+    dtype: torch.dtype = torch.float32,
+    low: Optional[float] = None,
+    high: Optional[float] = None,
+    requires_grad: bool = False,
+    noncontiguous: bool = False,
+    exclude_zero: bool = False
+) -> torch.Tensor:
+    r"""Creates a random tensor with the given :attr:`size`.
+
+    The function comes with other optional arguments to allow random tensor generation for the user's needs.
+
+    By default, :attr:`device` is ``cpu`` and :attr:`dtype` is ``torch.float32``, but user can pass these arguments
+    as required.
+
+    If :attr:`low` and :attr:`high` are not passed, following default values are considered depending on the given
+    :attr:`dtype`:
+
+    - boolean type: `low` = 0, `high` = 2.
+    - uint8 type: `low` = 0, `high` = 9.
+    - floating and integral types: `low` = -9 and `high` = 9.
+    - complex types, for each real and imaginary part: `low` = -9, `high` = 9.
+
+    If :attr:`low` and :attr:`high` are passed, they are considered only if they are within the
+    limit of the :attr:`dtype`. Following are a few conditions that are taken care of:
+
+    - If :attr:`low` and/or :attr:`high` are specified and within dtype limits: the values are taken as they were.
+    - If :attr:`low` and/or :attr:`high` are specified but exceed the limits: :attr:`dtype` limits ar considered instead.
+    - If :attr:`low` is ``-inf`` and/or :attr:`high` is ``inf``: :attr:`dtype` limits are considered instead
+    - If :attr:`low` is ``inf`` or ``nan`` and/or :attr:`high` is ``-inf`` or nan: a `ValueError` is raised, since these are invalid values for the range of output tensor.
+
+    If :attr:`noncontiguous` is ``True``, a non-contiguous tensor with the given size will be returned unless the
+    size specifies a tensor with a 1 or 0 elements in which case the non-contiguous parameter is ignored because
+    it is not possible to create a non-contiguous Tensor with a single element.
+
+    If :attr:`exclude_zero` is ``True`` (default is ``False``), all the values matching to zero in
+    the created tensor are replaced with a ``tiny`` (smallest positive representable number) value if floating type,
+    [``tiny`` + ``tiny``.j] if complex type and ``1`` if integer/boolean type.
+
+    Examples:
+        >>> import torch
+        >>> from torch.testing import make_tensor
+        >>> # Create a sample integral type (int64) tensor, default range = [-9, 9)
```

Comment:
"Create" -> "Creates"

The comment describes the code and the code CREATES a tensor.